File: Python/Controladores_bases/inicializador.py
#!/usr/bin/env python3

from Controladores_bases.Controlador import Cargo, Cliente, Empleado, Proveedor, EstadoServicio
from Controladores_bases.ACbase_usuario import base_usuario

usrdb = base_usuario()
cjsdb = 'cjs.db'

cargo_controller = Cargo(cjsdb)
cliente_controller = Cliente(cjsdb)
empleado_controller = Empleado(cjsdb)
proveedor_controller = Proveedor(cjsdb)
estado_controller = EstadoServicio(cjsdb)

def inicializar():
    # Cargo
    if not cargo_controller.obtener_cargo(1):
        id_cargo = cargo_controller.crear_cargo('user')
        print(f"Registro creado en 'cargo' con id: {id_cargo}")

    # Cliente
    if not cliente_controller.obtener_cliente(1):
        id_cliente = cliente_controller.crear_cliente('cliente', 'ninguno')
        print(f"Registro creado en 'cliente' con id: {id_cliente}")

    # Empleado
    if not empleado_controller.obtener_empleado(1):
        id_empleado = empleado_controller.crear_empleado('empleado', 1, 'ninguno')
        print(f"Registro creado en 'empleados' con id: {id_empleado}")

    # Estados de Servicio
    estados_default = ['recibido', 'en proceso', 'terminado']
    for i, estado in enumerate(estados_default, start=1):
        if not estado_controller.obtener_estado_servicio(i):
            id_estado = estado_controller.crear_estado_servicio(estado)
            print(f"Registro creado en 'estado_servicio' con id: {id_estado}")

    # Proveedor
    if not proveedor_controller.obtener_proveedor(1):
        id_proveedor = proveedor_controller.crear_proveedor('proveedor', 'ninguno', 'ninguno')
        print(f"Registro creado en 'proveedores' con id: {id_proveedor}")

    # Usuario
    if not usrdb.Consultar_usuario('admin'):
        user = usrdb.registrar_usuario('admin', 'admin')
        print(f"Registro creado en 'usuarios' con user: {user}")

if __name__ == "__main__":
    inicializar()

File: Flutter/Cjsapp/lib/Controladores/inicializador.dart
import 'package:cjsapp/Controladores/base_usuario_controlador.dart';
import 'package:cjsapp/Modelos/usuario.dart';
import 'package:cjsapp/Controladores/base_principal_controlador.dart';
import 'package:cjsapp/Modelos/principal.dart';
import 'dart:developer';

class Inicializacion {
  static final BaseUsuarioControlador _userDb = BaseUsuarioControlador();

  static Future<void> inicializar() async {
    await _crearUsuarioAdmin();
    await _crearCargoPorDefecto();
    await _crearClientePorDefecto();
    await _crearEmpleadoPorDefecto();
    await _crearEstadosServicioPorDefecto();
    await _crearProveedorPorDefecto();
  }

  static Future<void> _crearUsuarioAdmin() async {
    const String adminUsername = 'admin';
    const String adminPassword = 'admin';

    User? existingAdmin = await _userDb.consultarUsuario(adminUsername);
    if (existingAdmin == null) {
      User newAdmin = User(username: adminUsername, password: adminPassword);
      bool success = await _userDb.registrarUsuario(newAdmin);
      if (success) {
        log('Usuario admin creado con éxito');
      } else {
        log('No se pudo crear el usuario admin');
      }
    } else {
      log('El usuario admin ya existe');
    }
  }

  static Future<void> _crearCargoPorDefecto() async {
    Cargo? existingCargo = await DatabaseController.obtenerCargo(1);
    if (existingCargo == null) {
      Cargo newCargo = Cargo(cargo: 'user');
      int id = await DatabaseController.crearCargo(newCargo);
      log('Registro creado en "cargo" con id: $id');
    }
  }

  static Future<void> _crearClientePorDefecto() async {
    Cliente? existingCliente = await DatabaseController.obtenerCliente(1);
    if (existingCliente == null) {
      Cliente newCliente = Cliente(nombre: 'cliente', telefono: 'ninguno');
      int id = await DatabaseController.crearCliente(newCliente);
      log('Registro creado en "cliente" con id: $id');
    }
  }

  static Future<void> _crearEmpleadoPorDefecto() async {
    Empleado? existingEmpleado = await DatabaseController.obtenerEmpleado(1);
    if (existingEmpleado == null) {
      Empleado newEmpleado = Empleado(nombre: 'empleado', idCargo: 1, telefono: 'ninguno');
      int id = await DatabaseController.crearEmpleado(newEmpleado);
      log('Registro creado en "empleados" con id: $id');
    }
  }

  static Future<void> _crearEstadosServicioPorDefecto() async {
    List<String> estadosDefault = ['recibido', 'en proceso', 'terminado'];
    for (int i = 0; i < estadosDefault.length; i++) {
      EstadoServicio? existingEstado = await DatabaseController.obtenerEstadoServicio(i + 1);
      if (existingEstado == null) {
        EstadoServicio newEstado = EstadoServicio(descripcion: estadosDefault[i]);
        int id = await DatabaseController.crearEstadoServicio(newEstado);
        log('Registro creado en "estado_servicio" con id: $id');
      }
    }
  }

  static Future<void> _crearProveedorPorDefecto() async {
    Proveedor? existingProveedor = await DatabaseController.obtenerProveedor(1);
    if (existingProveedor == null) {
      Proveedor newProveedor = Proveedor(
        nombre: 'proveedor',
        telefono: 'ninguno',
        tipoDeProductos: 'ninguno'
      );
      int id = await DatabaseController.crearProveedor(newProveedor);
      log('Registro creado en "proveedores" con id: $id');
    }
  }
}